go/cmd/betabot/main.go

```diff
@@ -103,15 +105,71 @@ func handleEvent(ctx context.Context, messengerClient bertymessenger.MessengerSe
 			if err != nil {
 				return err
 			}
+		} else if contact.State == bertymessenger.Contact_Established {
+			// When contact was established, send message and a group invitation
+			time.Sleep(5 * time.Second)
```

Comment:
The event loop being monothread
https://github.com/berty/berty/blob/a39a7406523a8c83af81e730a336d3c142b9aa9b/go/cmd/betabot/main.go#L74-L76
You are effectively blocking the whole loop for 5 seconds.